src/create-extension-pack.js

```diff
@@ -0,0 +1,184 @@
+    if (!fs.existsSync(extPackSrcFolder)) {
+        await fs.mkdir(extPackSrcFolder);
+    }
+
+    const extPack = {};
+    extPack.name = packName;
+    extPack.displayName = packName;
+    extPack.description = 'Builtin extension pack associated to a version of vscode';
+    extPack.version = packVersion;
+    extPack.publisher = publisher;
+    extPack.categories = categories;
+    extPack.engines = { vscode: '^' + vscodeVer };
+    extPack.repository = repository;
+    extPack.extensionPack = await resolveExtensions();
+
+    if (extPack.extensionPack.length === 0) {
+        process.exitCode = 1;
+        console.error('Aborting: No extension was found available for this version: ' + packVersion);
+        return;
+    }
+
+    const packFolderPath = path.join(extPackSrcFolder, '..', extPackNameAndVersion)
+    const packJsonPath = path.join(packFolderPath, PACKAGE_JSON);
+    const licensePath = path.join(packFolderPath, 'LICENSE.txt');
+    const readmePath = path.join(packFolderPath, 'README.md');
+
+    fs.writeFileSync(packJsonPath, JSON.stringify(extPack, null, 2), 'utf-8');
+    console.log('Generated ' + PACKAGE_JSON + ' file at: ' + packJsonPath);
+    fs.writeFileSync(licensePath, generateLicense());
+    fs.writeFileSync(readmePath, generateReadme());
+
+    await vsce.createVSIX({
+        'cwd': packFolderPath,
+        'packagePath': dist(),
+        'useYarn': true
+    });
+
+    async function resolveExtensions() {
+        const extensionsArr = [];
+        for (const extension of fs.readdirSync(extensions())) {
+            const extDataPath = extensions(extension, PACKAGE_JSON);
+            if (!fs.existsSync(extDataPath)) {
+                console.log('No ' + PACKAGE_JSON + ' found for: ' + extension);
+                continue;
+            }
+
+            const content = fs.readFileSync(extDataPath, 'utf-8');
+            const extData = JSON.parse(content);
+
+            const extVsixPath = dist(extData.name + '-' + packVersion + '.vsix');
+            const available = await isAvailable(extVsixPath, extData.name, packVersion);
+            if (!available) {
+                console.log("Skipping extension, i.e. .vsix is not found and " +
+                    "neither published in the registry : " + extVsixPath);
+                continue;
+            }
+
+            const extensionId = extData.publisher + '.' + extData.name;
+            console.log('Adding: ' + extensionId);
+            extensionsArr.push(extensionId);
+        }
+        return Promise.resolve(extensionsArr);
+    }
+
+    async function isAvailable(extVsixPath, extensionName, extensionVersion, namespace = 'vscode') {
+        if (fs.existsSync(extVsixPath)) {
+            return Promise.resolve(true);
+        }
+
+        return isPublished(extensionVersion, extensionName, namespace);
+    }
+})();
+
+function generateLicense() {
+    const date = new Date();
+    const year = date.getFullYear();
+    return `MIT License
```

Comment:
For the license of the generated extension packs, we should probably go with our project's license (EPL2 + GPL classpath exception)?

Reply:
a bit more details:

The vscode extension pack we're generating is a collection of "base" extensions, that we think are useful for Theia-based applications. As per the syntax of such a pack, the extensions are not contained but referenced, by id and namespace. 

In this case the extensions are MIT-licensed. But the pack(s), being our creation, can be under another license, I think. And by default, anything we produce should be under the project's license, wherever it's possible. 

WDYT?

Reply:
@marcdumais-work I think our project license (EPL2 + GPL) would make the most sense, these are our packs like you've mentioned, where we reference the extensions we want by `id`.

Reply:
@marcdumais-work, @vince-fugnitto , When generating an extension pack for vscode (using yo code) we can see that no LICENSE file gets generated, as seen in the file structure below.
 I think no LICENSE in the generated pack is appropriate as the extension pack is auto generated so there is no real human behind it, then the applicable License(s) should be the ones stated on each extension. Including one LICENSE for the pack would be misleading and the consumers may think the stated LICENSE applies to all extensions which is not the case.

What do you think about excluding the LICENSE file altogether ? 

![image](https://user-images.githubusercontent.com/76971376/117314778-1607df80-ae55-11eb-9b10-50a22330dfe3.png)

Reply:
I had an offline discussion with Vince and it seems like including a License is expected by the foundation so I am leaving it as suggested using EPL2+GPL.